File: Arrays & HashMap/Contains Duplicate/Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return False
        
        map = defaultdict(int)
        for num in nums:
            if num in map:
                return True
            map[num] = True

        return False

'''
Time Complexity: O(n)
Space Complexity: O(n)
'''
            

File: Arrays & HashMap/Contains Duplicate/Contains Duplicate.kt
class Solution {
    fun containsDuplicate(nums: IntArray): Boolean {
        if (nums.size == 1) return false

        val map = mutableMapOf<Int, Boolean>()

        nums.forEach { num -> 
            if (map.contains(num)) return true
            else {
                map[num] = true
            }
        }
        return false
    }
}

//Time Complexity: O(n)
//Space Complexity: O(n)